compiler/passes/scopeResolve.cpp

```diff
@@ -2779,6 +2779,24 @@ static void lookupAndAddToVisibleMap(const char* name, CallExpr* call,
 static void processGetVisibleSymbols() {
   forv_Vec(CallExpr, call, gCallExprs) {
     if (call->isPrimitive(PRIM_GET_VISIBLE_SYMBOLS)) {
+      bool ignoreInternalModules = false;
+      // look for a single NamedExpr argument ignoreInternals=true|false
+      if (call->numActuals() == 1) {
+        NamedExpr* ne = toNamedExpr(call->get(1));
+        if (ne && !strcmp(ne->name, "ignoreInternalModules")) {
+          SymExpr* se = toSymExpr(ne->actual);
+          if (se && (se->symbol() == gTrue || se->symbol() == gFalse)) {
+            ignoreInternalModules = se->symbol() == gTrue;
+          } else {
+            USR_FATAL(se, "get visible symbols only argument must be a literal 'true' or 'false'");
+          }
+        } else {
+          USR_FATAL(call, "get visible symbols only argument must be a named expression named ignoreInternalModules");
+        }
+      } else {
+        if (call->numActuals() != 0)
+          USR_FATAL(call, "get visible symbols may only have 0 or 1 arguments");
```

Comment:
I'd like a test of these error messages :)

Reply:
Done. Thanks!